metadeploy/api/salesforce.py

```diff
@@ -60,29 +96,36 @@ def _get_org_details(*, cci, org_name, project_path):
     return (scratch_org_config, scratch_org_definition)
 
 
-def _refresh_access_token(*, config, org_name):
+def _refresh_access_token(*, scratch_org, config, org_name, keychain=None):
     """Refresh the JWT.
 
     Construct a new OrgConfig because ScratchOrgConfig tries to use sfdx
     which we don't want now -- this is a total hack which I'll try to
     smooth over with some improvements in CumulusCI
     """
-    org_config = OrgConfig(config, org_name)
-    org_config.refresh_oauth_token = Mock()
-    info = jwt_session(
-        SF_CLIENT_ID, SF_CLIENT_KEY, org_config.username, org_config.instance_url
-    )
-    org_config.config["access_token"] = info["access_token"]
-    return org_config
+    try:
+        org_config = OrgConfig(config, org_name, keychain=keychain)
+        org_config.refresh_oauth_token = Mock()
+        info = jwt_session(
+            SF_CLIENT_ID, SF_CLIENT_KEY, org_config.username, org_config.instance_url
+        )
+        if info != org_config.config:
+            org_config.config.update(info)
+        org_config._load_userinfo()
+        org_config._load_orginfo()
```

Comment:
@davisagli I tried replacing this with `org_config.refresh_oauth_token(None)` (a la https://github.com/oddbird/MetaDeploy/pull/439#discussion_r519016033 and https://github.com/oddbird/MetaDeploy/pull/498#discussion_r527739165), but it caused a "could not decode key" error. I'm pretty sure it would work in production, but fails via Docker because of how we're setting the `SFDX_HUB_KEY` env var (see https://github.com/moby/moby/issues/12997). This works, and still seems improved.

Reply:
Yeah -- for docker it would probably be best to provide the encoded key in a different variable, and have the start_server.sh script decode it and re-export as SFDX_HUB_KEY since CumulusCI expects the environment variable to contain a key that is not encoded